core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -918,19 +918,19 @@ public SystemSessionProperties(
                         FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT,
                         "Maximum number of partitions for distributed joins and aggregations executed with fault tolerant execution enabled",
                         queryManagerConfig.getFaultTolerantExecutionMaxPartitionCount(),
-                        value -> validateIntegerValue(value, FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT, 1, false),
+                        value -> validateIntegerValue(value, FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT, 1, 1000, false),
```

Comment:
nit: extract constant for 1000

Reply:
Also I am not a big fan of hardcoding 1000. It is a bit arbitrary.
Probably it should be limitted to `queryManagerConfig.getFaultTolerantExecutionMaxPartitionCount()` and we would need an extra config parameter to set default value of session property.

But I do not feel to strongly about that.

Reply:
It's weird to have too many configs:) I don't see any similar configs like that, so I extract constant (similar to `MAX_TASK_RETRY_ATTEMPTS`)

Reply:
`MAX_TASK_RETRY_ATTEMPTS` is different animal. As with that you just set number of attempts. And here you specify `FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT` you specify number of partitions and `1000` hard coded limit on what you can set. There is no such hardcoded limit for `MAX_TASK_RETRY_ATTEMPTS`. 

Let's leave 1000 in the code for now. We can revisit later if needed.